flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTable.java

```diff
@@ -242,10 +263,15 @@ public boolean allocateSlot(int index, JobID jobId, AllocationID allocationId, T
 			return false;
 		}
 
+		ResourceProfile slotResourceProfile = defaultSlotResourceProfile;
+		if (!resourceProfileBookkeeper.reserve(slotResourceProfile)) {
+			return false;
```

Comment:
would be nice to log how much we tried to allocate and what was available budget in case of failure